comp/metadata/runner/runner.go

```diff
@@ -0,0 +1,120 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package runner
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"go.uber.org/fx"
+)
+
+type MetadataProvider func(context.Context) time.Duration
```

Comment:
I guess that `time.Duration` defines the interval duration between 2 calls. Are there cases where this value is dynamic?
I would tend to pass extra information about the provider as a struct because it is more flexible and more explicit about the meaning of `time.Duration`
```go
type MetadataProvider struct {
   callback func(context.Context) 
   interval time.Duration
}
```

Reply:
Yes some providers have dynamic interval base on data updates or if it's their first run or not.